[RAI-19749] GC scheduler refinements (#52294)

d-netto opened the pull request:
## PR Description

Backports https://github.com/JuliaLang/julia/pull/52294.

Requirements for merging:
- [x] I have opened an issue or PR upstream on JuliaLang/julia: https://github.com/JuliaLang/julia/pull/52294
- [x] I have removed the `port-to-*` labels that don't apply.
- [x] I have opened a PR on raicode to test these changes: https://github.com/RelationalAI/raicode/pull/16625

d-netto commented:
Bump @kpamnany.

kpamnany commented:
There seems to be a clear [regression](https://g-925fbbc735.grafana-workspace.us-east-1.amazonaws.com/d/hydra-overview/overview?orgId=1&var-base_jobset=master&var-diff_jobsets=dcn-new-gc-scheduler) for MMI. It'd be good to understand why?

d-netto commented:
This dashboard that you linked refers to the old commit, which was on top of the 4k pages PR. The regression there is unsurprising because of this.

Given that we reverted this 4k pages commit, I've rebased the branch and pushed a new commit to raicode. We should wait to see what the results are with 16k pages, which will allows us to better isolate the performance effects from this patch.

kpamnany reviewed (approved):
Approving to remove the blocker but please confirm the benchmark perf before merging.
